src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -37,8 +39,23 @@ public JsonLineNumberResolver(TemplateContext templateContext)
         /// or 1 if it can't be determined.</returns>
         public int ResolveLineNumber(string pathInExpandedTemplate)
         {
-            JToken expandedTemplateRoot = this.templateContext.ExpandedTemplate;
-            JToken originalTemplateRoot = this.templateContext.OriginalTemplate;
+            var rootTemplateContext = this.templateContext;
+            while (rootTemplateContext != null && !rootTemplateContext.IsMainTemplate)
+            {
+                rootTemplateContext = rootTemplateContext.ParentContext;
+            }
+
+            if (rootTemplateContext == null)
+            {
+                throw new ArgumentNullException(nameof(rootTemplateContext));
```

Comment:
I think these two exceptions should be of a different type (maybe TemplateAnalyzerExceptions) and give a bit more information, like "Main template was not found"

Reply:
I couldn't get the TemplateAnalyzerException to work in this specific file and it doesn't autosuggest an import. Any thoughts @JohnathonMohr

Reply:
TemplateAnalyzerException is defined in the .Core project, so it's not accessible here.  However, ArgumentNullException is used for referring to an argument passed to a function, which is not the scenario here.  For now just throw a regular Exception.